lib/1.4/dml-builtins.dml

```diff
@@ -2262,53 +2311,33 @@ template bank is (object, shown_desc) {
         } else {
             if (SIM_mem_op_is_read(memop)) {
                 local uint64 value = 0;
-                local bool inquiry_override = false;
-                _callback_before_read(
-                    this._bank_obj(), ini, &inquiry_override,
-                    &offset, size, &_connections, &_before_read_callbacks);
                 try {
-                    if (inquiry_override) {
-                        value = this.get(offset, size);
-                    } else {
-                        value = this.read(offset, _mask(size), aux);
-                    }
+                    value = this.read(offset, _mask(size), aux);
                 } catch {
                     success = false;
                 }
-                _callback_after_read(this._bank_obj(), ini, &offset, size,
-                                     &value, &success, &_connections,
-                                     &_after_read_callbacks);
 
                 if (success) {
                     this._memop_set_read_value(memop, value);
                 }
             } else {
                 local uint64 writevalue = this._memop_write_value(memop);
-
-                local bool suppress = false;
-                _callback_before_write(
-                    this._bank_obj(), ini, &offset, size, &writevalue,
-                    &suppress, &_connections, &_before_write_callbacks);
-                if (!suppress) {
-                    try {
-                        this.write(offset, writevalue, _mask(size), aux);
-                    } catch {
-                        success = false;
-                    }
+                try {
+                    this.write(offset, writevalue, _mask(size), aux);
+                } catch {
+                    success = false;
                 }
-                _callback_after_write(this._bank_obj(), ini, &offset, size,
-                                      &success, &_connections,
-                                      &_after_write_callbacks);
             }
         }
 
+        dev._current_initiator = prev_initiator;
         return success;
     }
 
     shared method transaction_access(transaction_t *t, uint64 offset,
                                      void *aux) -> (exception_type_t) default {
         local uint64 size = SIM_transaction_size(t);
-        local conf_object_t *ini = SIM_transaction_initiator(t);
+        local conf_object_t *ini = dev._current_initiator;
```

Comment:
This surprises me. We don't need `SIM_transaction_initiator` nor `SIM_get_mem_op_initiator` at all anymore? Their results will always be equal to the object we get in `_transaction_access`?

Reply:
Oh, I think this hunk should be dropped, apparently our tests don't seem to cover initiator.